OpenRA.Mods.Common/Traits/SpawnActorOnDeath.cs

```diff
@@ -75,38 +81,43 @@ public void Killed(Actor self, AttackInfo e)
 			if (info.DeathType != null && !e.Damage.DamageTypes.Contains(info.DeathType))
 				return;
 
-			self.World.AddFrameEndTask(w =>
-			{
-				// Actor has been disposed by something else before its death (for example `Enter`).
-				if (self.Disposed)
-					return;
-
-				var td = new TypeDictionary
-				{
-					new ParentActorInit(self),
-					new LocationInit(self.Location),
-					new CenterPositionInit(self.CenterPosition),
-					new FactionInit(faction)
-				};
-
-				if (info.OwnerType == OwnerType.Victim)
-					td.Add(new OwnerInit(self.Owner));
-				else if (info.OwnerType == OwnerType.Killer)
-					td.Add(new OwnerInit(e.Attacker.Owner));
-				else
-					td.Add(new OwnerInit(self.World.Players.First(p => p.InternalName == info.InternalOwner)));
+			// Actor has been disposed by something else before its death (for example `Enter`).
+			if (self.Disposed)
```

Comment:
I don't really see the need to move all this code, either. In my opinion the code would be both safer and much easier to review if only the delay-relevant lines were added to the end of the FrameEndTask and the rest was left untouched as far as possible.

If there's a specific reason for the move, please explain, otherwise I'd request to move everything back into the FrameEndTask.